projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_error_handler.py

```diff
@@ -441,3 +444,78 @@ def _is_pool_error(exception: Exception) -> bool:
             classname_with_package=".*OperationalError.*",
             exception_message_matcher_regex=".*admission for timeout ms in pool.*",
         )
+
+    def _handle_memory_error(self, exception, recovery_cursor):
```

Comment:
This is pretty long logic . We are adding more than 100 lines. I think it would be better to encapsulate the logic in separate class OutOfMemoryErrorHandler for example   . And here call it with 2 or 3 lines onlyh.

There you can also write a good nice pydoc in that new class explaining all the intricacies and the whys and the whats of your solution.

Reply:
Done.